Code/autopkglib/__init__.py

```diff
@@ -29,6 +30,43 @@
 
 import FoundationPlist
 
+
+def memoize(func):
+    """
+    Decorator to memoize results of function calls.
+    Works with functions that have no args.
+    https://www.thumbtack.com/engineering/a-primer-on-python-decorators/
+    """
```

Comment:
why not use an implementation that works with functions that _have_ args, too?

https://github.com/munki/munki/blob/master/code/client/munkilib/utils.py#L33-L44

Reply:
Awesome, I'll use that instead.